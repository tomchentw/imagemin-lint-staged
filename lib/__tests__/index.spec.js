import fs from "fs";
import child_process from "child_process";
import promisify from "util.promisify";

import { minifyFile } from '../index'

const stat = promisify(fs.stat);
const exec = promisify(child_process.exec);

describe("index module", () => {
  const FILENAMES = [
    "./lib/__fixtures__/test.gif",
    "./lib/__fixtures__/test.jpg",
    "./lib/__fixtures__/test.png",
    "./lib/__fixtures__/test.svg",
  ];

  const stats = () =>
    Promise.all(
      FILENAMES.map(async (f) => {
        const { size } = await stat(f);
        return { f, size };
      })
    );


  describe("minifyFile function", () => {
    it("should work as expected", async () => {
      const before = await stats();
      await Promise.all(FILENAMES.map(minifyFile));
      const after = await stats();

      await exec(`git checkout .`);
      expect(after).not.toEqual(before);

      expect(before).toMatchSnapshot();
      expect(after).toMatchSnapshot();
    });
  });
});
