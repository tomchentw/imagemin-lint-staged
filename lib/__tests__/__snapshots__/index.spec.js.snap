// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index module minifyFile function should work as expected 1`] = `
Array [
  Object {
    "f": "./lib/__fixtures__/test.gif",
    "size": 78080,
  },
  Object {
    "f": "./lib/__fixtures__/test.jpg",
    "size": 50986,
  },
  Object {
    "f": "./lib/__fixtures__/test.png",
    "size": 71834,
  },
  Object {
    "f": "./lib/__fixtures__/test.svg",
    "size": 25118,
  },
]
`;

exports[`index module minifyFile function should work as expected 2`] = `
Array [
  Object {
    "f": "./lib/__fixtures__/test.gif",
    "size": 80201,
  },
  Object {
    "f": "./lib/__fixtures__/test.jpg",
    "size": 45131,
  },
  Object {
    "f": "./lib/__fixtures__/test.png",
    "size": 61692,
  },
  Object {
    "f": "./lib/__fixtures__/test.svg",
    "size": 20698,
  },
]
`;
